src/main/java/webs/hillel/Cart.java

```diff
@@ -0,0 +1,59 @@
+package webs.hillel;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+@Component
+public class Cart {
+    private List<Product> products;
+    @Autowired
+    public Cart() {
+        products = new ArrayList<>();
+    }
```

Comment:
What are you trying to autowire? The collection is created on a class level, and that's it, no injection was made for this class.